File: 백준/Silver/1543. 문서 검색/문서 검색.cc
#include <bits/stdc++.h>

using namespace std;

int main() {
    string line;
    string find;

    getline(cin, line); getline(cin, find);

    int s = 0;
    int cnt = 0;
    while(true) {
        if(line.find(find) != string::npos) {
            int idx = line.find(find);
            line = line.substr(idx+ find.length());
            cnt++;
        }else
            break;
    }

    cout <<cnt;

}

File: 백준/Silver/1543. 문서 검색/문서 검색.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str = sc.nextLine();
        String find = sc.nextLine();

        int cnt = 0;
        while (true) {
            int idx = str.indexOf(find);
            if (idx == -1) {
                break;
            }
            cnt++;
            str = str.substring(idx + find.length());
        }

        System.out.println(cnt);
    }
}